src/db.rs

```diff
@@ -83,6 +83,7 @@ pub struct WriteBatch {
 
 pub struct ReadOptions {
     inner: *mut ffi::rocksdb_readoptions_t,
+    iterate_upper_bound: Vec<u8>,
```

Comment:
It seems that `Option<Vec<u8>>` would be better choice ?

Reply:
Sounds reasonable to me.